framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -127,7 +127,13 @@ PenetrationLocator::detectPenetration()
 void
 PenetrationLocator::reinit()
 {
+  // Delete the PenetrationInfo objects we own before clearing the
+  // map, or we have a memory leak.
+  for (auto & it : _penetration_info)
+    delete it.second;
```

Comment:
Holy dooley! I think we should switch these to smart pointers, in this case, keeping the `unique_ptr`around should be fine since these are stored in a standard map. This is a good fix, so I'll open a ticket to make that change later.